types/event-kit/index.d.ts

```diff
@@ -83,31 +83,40 @@ export class Emitter<Emissions = { [key: string]: any }> implements DisposableLi
 
     // Event Subscription
     /** Registers a handler to be invoked whenever the given event is emitted. */
-    on<T extends keyof Emissions>(eventName: T, handler: (value?: Emissions[T]) => void):
+    on<T extends keyof Emissions>(eventName: T, handler: (value: Emissions[T]) => void):
         Disposable;
 
     /**
      *  Register the given handler function to be invoked the next time an event
      *  with the given name is emitted via ::emit.
      */
-    once<T extends keyof Emissions>(eventName: T, handler: (value?: Emissions[T]) => void):
+    once<T extends keyof Emissions>(eventName: T, handler: (value: Emissions[T]) => void):
         Disposable;
 
     /**
      *  Register the given handler function to be invoked before all other
      *  handlers existing at the time of subscription whenever events by the
      *  given name are emitted via ::emit.
      */
-    preempt<T extends keyof Emissions>(eventName: T, handler: (value?: Emissions[T]) => void):
+    preempt<T extends keyof Emissions>(eventName: T, handler: (value: Emissions[T]) => void):
         Disposable;
 
     // Event Emission
     /** Invoke the handlers registered via ::on for the given event name. */
-    emit<T extends keyof Emissions>(eventName: T, value?: Emissions[T]): void;
+    emit<T extends keyof Emissions>(eventName: T, value: Emissions[T]): void;
+
+    /** Invoke the handlers registered via ::on for the given event name. */
+    emit(eventName: string): void;
```

Comment:
Shouldn't these be also keyed by `keyof Emissions` though?

Reply:
Yes. Now that I think about it, we can just keep the old signature here and get rid of the overload. This accomplishes exactly the same thing.